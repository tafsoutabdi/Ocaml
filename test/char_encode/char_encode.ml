open Tokenization.Characters.Tokenizer
open Utils.Characters
open Utils.Std

let print_context voc s = 
  Format.printf "voc = %s@.s = \"%s\"@.encode voc s = "
    (string_of_vocabulary voc) s

let () = 
  let voc = ['a'; 'b'; 'r'; 'c'; 'd'] in
  let s = "abracadabra" in
  print_context voc s;
  print_id_list (encode voc s);
  Format.printf "@."

let () = 
  let voc = ['a'; 'b'; 'r'; 'c'; 'd'] in
  let s = "arbre barbara" in
  print_context voc s;
  try ignore @@ encode voc s
  with EncodingError s -> 
    Format.printf "EncodingError \"%s\"@." s 