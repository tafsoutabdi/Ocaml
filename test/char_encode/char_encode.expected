voc = ['a'; 'b'; 'r'; 'c'; 'd']
s = "abracadabra"
encode voc s = [0; 1; 2; 0; 3; 0; 4; 0; 1; 2; 0]
voc = ['a'; 'b'; 'r'; 'c'; 'd']
s = "arbre barbara"
encode voc s = EncodingError "e barbara"
